ddcommon/src/profiled_endpoints.rs

```diff
@@ -0,0 +1,72 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
```

Comment:
I'm curious about the decision to add this new feature to common, rather than under profiling. Did you consider having it all under profiling? Is there any constraint that makes it hard to do so?

Reply:
Good question.
No constraint, to be honest I'm not used to the folder structure and I hope this PR would help in getting more insights :)

Reply:
✔️